src/js/components/homepage/visualizations/TreeMap.jsx

```diff
@@ -84,29 +96,90 @@ export default class TreeMap extends React.Component {
                 y1={n.y1}
                 total={n.parent.value}
                 key={i}
-                color={colors[i]} />
+                color={colors[i]}
+                chosen={this.state.category}
+                toggleTooltip={this.toggleTooltip}
+                showOverlay={this.state.showOverlay} />
         );
 
         this.setState({
             finalNodes: nodes
         });
     }
 
+    toggleTooltip(cat, value) {
+        const descSet = this.state.descriptions;
+        // find index of object item on matching cat name
+        let descIndex = '0';
+        if (cat !== 'none') {
+            descIndex = _.findIndex(descSet, { name: cat });
+        }
+
+        // set it to desc value
+        let desc = '';
+        if (cat !== 'none') {
+            desc = descSet[descIndex].value;
+        }
+
+        // set the state
+        this.setState({
+            category: cat,
+            description: desc,
+            individualValue: value,
+            showOverlay: false
+        });
+        const newNodes = [];
+
+        // create a copy of finalNodes
+        // update the copy
+        this.state.finalNodes.forEach((i) => {
+            const originalNode = i;
+            const updatedNode = update(originalNode, {
```

Comment:
In my original approach, I wanted to re-run the `buildTree` function, but was concerned that running the entire function again would be too heavy.  Additionally, as the overlay state changes, it changes in `state`, so I wasn't sure that `componentWillReceiveProps` was appropriate.  Would an update in `componentDidUpdate` be appropriate?